tests/rkt_image_list_test.go

```diff
@@ -0,0 +1,178 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/steveeJ/gexpect"
+)
+
+type ImageId struct {
+	path string
+	hash string
+}
+
+func (imgId *ImageId) getShortHash(length int) (string, error) {
+	hashLen := 1 + length
+	if hashLen > len(imgId.hash) {
+		return "", fmt.Errorf("getShortHash: Hash %s is shorter than %d chars", imgId.hash, hashLen)
+	}
+
+	return imgId.hash[:hashLen], nil
+}
+
+// containsConflictingHash returns an ImageId pair if a conflicting short hash is found. The minimum
+// hash of 2 char is used for comparisons.
+func (imgId *ImageId) containsConflictingHash(imgIds []ImageId) (imgIdPair []ImageId, found bool) {
+	shortHash, err := imgId.getShortHash(2)
+	if err != nil {
+		panic(fmt.Sprintf("containsConflictingHash: %s", err))
+	}
+
+	for _, iId := range imgIds {
+		if strings.HasPrefix(iId.hash, shortHash) {
+			imgIdPair = []ImageId{*imgId, iId}
+			found = true
+			break
+		}
+	}
+	return
+}
+
+// TestShortHash tests that the short hash generated by the rkt image list
+// command is usable by the commands that accept image hashes.
+func TestShortHash(t *testing.T) {
+	var (
+		imageIds []ImageId
+		iter     int
+	)
+
+	for {
+		image := patchTestACI(fmt.Sprintf("rkt-shorthash-%d.aci", iter), fmt.Sprintf("--name=shorthash--%d", iter))
+		defer os.Remove(image)
+
+		imageHash := getHashOrPanic(image)
+		imageId := ImageId{image, imageHash}
+
+		imageIdPair, isMatch := imageId.containsConflictingHash(imageIds)
+		if isMatch {
+			imageIds = imageIdPair
+			break
+		} else {
+			imageIds = append(imageIds, imageId)
+			iter++
+		}
+	}
+	ctx := newRktRunCtx()
+	defer ctx.cleanup()
+
+	// Pull images into cas
+	for _, imageId := range imageIds {
+		cmd := fmt.Sprintf("%s --insecure-skip-verify fetch %s", ctx.cmd(), imageId.path)
+		t.Logf("Fetching %s: %v", imageId.path, cmd)
+		child, err := gexpect.Spawn(cmd)
+		if err != nil {
+			t.Fatalf("Cannot exec: %v", err)
+		}
+		if err := child.Wait(); err != nil {
+			t.Fatalf("rkt didn't terminate correctly: %v", err)
+		}
+	}
+
+	// Get hash from 'rkt image list'
+	imageListCmd := fmt.Sprintf("%s image list --fields=key --no-legend", ctx.cmd())
+	child, err := gexpect.Spawn(imageListCmd)
+	if err != nil {
+		t.Fatal("Cannot exec rkt image list")
+	}
+
+	hash := fmt.Sprintf("sha512-%s", imageIds[0].hash[:12])
+	if err = expectWithOutput(child, hash); err != nil {
```

Comment:
The duplicate hash should be printed twice by `rkt image list`, right? It could be expected twice.